pico_w/wifi_vsys/wifi_vsys.cpp

```diff
@@ -0,0 +1,162 @@
+        printf("failed to connect.\n");
+        return 1;
+    } else {
+        printf("Connected.\n");
+    }
+
+    //set http client handlers
+    settings.result_fn = httpresult;
+    settings.headers_done_fn = headers;   
+    //define IP address that hosts the web api to send the data
+    IP4_ADDR(&ip, 192, 168, 0, 1);
+    int ledStatus = 0;
+    float vsys=0;
+
+    while(true) {        
+        //flip onboard led
+        ledStatus = 1 - ledStatus;
+        cyw43_arch_gpio_put(CYW43_WL_GPIO_LED_PIN, ledStatus);
+
+        //Select ADC2 to read moisture sensor
+        adc_select_input(2);
+        uint16_t result = adc_read();
+        float voltage = result * conversion_factor;
+        printf("ADC2 value: 0x%03x, %i, voltage: %3.1f V\n", result, result, voltage);
+
+        //make sure there is not HTTP request in progress or other wifi activity when we want to check the battery status    
+        if (!httpBusy){            
+            httpBusy = true;
+            
+            //prevent wifi background processing so we can use the shared pins
+            cyw43_thread_enter();   //or macro: CYW43_THREAD_ENTER
+
+            //WL_GPIO2, IP VBUS sense - high if VBUS is present, else low
+            uint vbus = cyw43_arch_gpio_get(CYW43_WL_GPIO2); 
+
+            //we could decide to only check the voltage if not powered by USB port/VBUS and perhaps not check it every cycle.
+            //if (vbus==0)            
+            //{
+            //initialize and select ADC3/GPIO29                
+            adc_gpio_init(29);
+            adc_select_input(3);
+
+            //Read ADC3, result is 1/3 of VSYS, so we still need to multiply the conversion factor with 3 to get the input voltage
+            uint16_t adc3 = adc_read();
+            vsys = adc3 * conversion_factor * 3; 
+            printf("ADC3 value: 0x%03x, %i, voltage: %f V\n", adc3, adc3, vsys);
+            //set GPIO29 back to the settings for wifi, otherwise cyw43 will not work correctly
+            SetGPIO29WifiStatus();
+            //}
+            //else{                
+                //USB/VSYS powered, do nothing (voltage should be somewhere around 5V)
+            //}
+            //exit to allow wifi communication again
+            cyw43_thread_exit();    //or macro: CYW43_THREAD_EXIT
+
+            sprintf(apicall, "/api/echo?m=%3.3f&v=%3.3f&b=%i", voltage, vsys, vbus);
+            printf("Calling API: %s\n", apicall);
+            err_t err = httpc_get_file(
+                &ip,        //IP address hosting the API
+                5131,       //port
+                apicall,    //api call
+                &settings,  //handlers for result/headers
+                body,       //handler for body
+                NULL,
+                NULL
+            );  
+            printf("[1] http status %d \n", err);
+        }
+        else{
+            printf("[1] http client is busy...\n");
```

Comment:
What are all these `[1]` supposed to indicate?

Reply:
ah, some leftover from debugging, totally irrelevant

Reply:
leftover, as in, I was running the http request on core1 initially, so it was indicating from which core the printf was coming...